File: src/org/lineageos/setupwizard/DeviceSpecificActivity.java
/*
 * SPDX-FileCopyrightText: 2021-2024 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.setupwizard;

import android.content.ComponentName;
import android.content.Intent;

import org.lineageos.setupwizard.util.SetupWizardUtils;

public class DeviceSpecificActivity extends SubBaseActivity {

    private static final String ACTION_SETUP_DEVICE = "org.lineageos.settings.device.SUW_SETTINGS";

    protected void onStartSubactivity() {
        Intent intent = new Intent(ACTION_SETUP_DEVICE);
        ComponentName name = intent.resolveActivity(getPackageManager());
        if (name != null) {
            startSubactivity(intent);
        } else {
            SetupWizardUtils.disableComponent(this, DeviceSpecificActivity.class);
            finishAction(RESULT_OK);
        }
    }
}

File: app/src/main/java/org/lineageos/setupwizard/system/DeviceSpecificActivity.kt
/*
 * SPDX-FileCopyrightText: The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.setupwizard.system

import android.content.Intent
import org.lineageos.setupwizard.base.SubBaseActivity
import org.lineageos.setupwizard.util.SetupWizardUtils

class DeviceSpecificActivity : SubBaseActivity() {

    override fun onStartSubactivity() {
        val intent = Intent(ACTION_SETUP_DEVICE)
        if (intent.resolveActivity(packageManager) != null) {
            startSubactivity(intent)
        } else {
            SetupWizardUtils.disableComponent(this, DeviceSpecificActivity::class.java)
            finishAction(RESULT_OK)
        }
    }

    companion object {
        private const val ACTION_SETUP_DEVICE = "org.lineageos.settings.device.SUW_SETTINGS"
    }
}
